gems/pending/appliance_console/date_time_configuration.rb

```diff
@@ -0,0 +1,96 @@
+require "appliance_console/logging"
+require 'appliance_console/prompts'
+
+module ApplianceConsole
+  class DateTimeConfiguration
+    DATE_REGEXP   = /^(2[0-9]{3})-(0?[1-9]|1[0-2])-(0?[1-9]|[12][0-9]|3[01])/
+    DATE_PROMPT   = "current date (YYYY-MM-DD)".freeze
+    TIME_REGEXP   = /^(0?[0-9]|1[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])/
+    TIME_PROMPT   = "current time in 24 hour format (HH:MM:SS)".freeze
+
+    attr_accessor :new_date, :new_time
+
+    include ApplianceConsole::Logging
+
+    def initialize(config = {})
+      @new_date = config[:new_date]
+      @new_time = config[:new_time]
+    end
+
+    def activate
+      say("Applying time configuration...")
+      disable_auto_sync &&
+        configure_date_time
```

Comment:
Thank you for the reply @Fryguy and @kbrock.
The chaining is to avoid invoking `configure_date_time` if `disable_auto_sync` fails.

I thought it was clean and easy to follow but am totally open to any suggestions.